cmd/azure/azure.go

```diff
@@ -88,15 +113,93 @@ func run() error {
 			Hyperthreading: types.HyperthreadingEnabled,
 		},
 		Platform: types.Platform{
-			Azure: &azure.Platform{},
+			Azure: &azure.Platform{
+				Region: ec.Region,
+			},
 		},
 		PullSecret: string(pullSecret),
 	}
+	return &cfg, nil
+}
+
+func run() error {
+	rootCmd := &cobra.Command{
+		Use:  "./azure [component]",
+		Long: "Azure Red Hat OpenShift dispatcher",
+	}
+	rootCmd.PersistentFlags().StringP("loglevel", "l", "Debug", "Valid values are [Debug, Info, Warning, Error]")
+	rootCmd.PersistentFlags().StringP("name", "n", "demo-cluster", "Cluster name value")
+	rootCmd.Printf("gitCommit %s\n", gitCommit)
+	rootCmd.Execute()
 
+	name, err := rootCmd.Flags().GetString("name")
+	if err != nil {
+		return err
+	}
+
+	cfg, err := getInstallConfig(name)
+	if err != nil {
+		return errors.Wrap(err, "failed to get InstallConfig")
+	}
+
+	defaults.SetInstallConfigDefaults(cfg)
+	if err := validation.ValidateInstallConfig(cfg, openstackvalidation.NewValidValuesFetcher()).ToAggregate(); err != nil {
+		return errors.Wrap(err, "invalid install config")
+	}
 	fmt.Println(cfg)
 
+	data, err := yaml.Marshal(cfg)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal InstallConfig")
+	}
+
+	// doing this to prevent the stdin questions.
+	ic := &installconfig.InstallConfig{}
+	ic.Config = cfg
+	ic.File = &asset.File{
+		Filename: "install-config.yaml",
+		Data:     data,
+	}
+	waIc := []asset.WritableAsset{ic}
+
 	// TODO: Start consuming assets. Assets need to be file system agnostic
```

Comment:
I am not sure what you mean above..

for the fakerp, i'd expect to use store.NewStore() and store like the current fakerp does and the v4 installer. however for the realrp, i'd expect we need to re-implement the Store API so that it stores this stuff in a DB like MSFT does currently.

Reply:
This is what I had in mind. Its was more metal note for myself.